src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -50,6 +59,35 @@ object EntityClient {
     }
   }
 
+  //returns (contents of participants.tsv, contents of samples.tsv)
+  def unzipTSVs(zipFile: ZipFile): (Option[String], Option[String]) = {
+    val zipEntries = zipFile.entries.asScala
+
+    val rand = java.util.UUID.randomUUID.toString.take(8)
+    val participantsFn = s"/tmp/$rand-participants.tsv"
+    val samplesFn = s"/tmp/$rand-samples.tsv"
+
+    val unzippedFiles = zipEntries.foldLeft((None: Option[String], None: Option[String])){ (acc: (Option[String], Option[String]), ent: ZipEntry) =>
+      if(!ent.isDirectory && ent.getName.contains("participants.tsv")) {
```

Comment:
does this code only look one directory down? it could be nested under several directories

Reply:
as far as i can tell from the docs it's flat, not recursive, but i'll doublecheck

Reply:
sorry, i meant that the list of entries (i.e. the variable `zipEntries` here) is a list, not some kind of tree structure. so walking that list traverses all files in the zip, including in subdirectories, because the tree structure has been flattened

Reply:
great - thanks!